src/LinAlg/hiopLinAlgFactory.cpp

```diff
@@ -204,22 +204,36 @@ hiopMatrixSparse* LinearAlgebraFactory::createMatrixSymSparse(int size, int nnz)
 /**
  * @brief Static method to create a raw C array
  */
-double* LinearAlgebraFactory::createRawArray(int n)
+double* LinearAlgebraFactory::createRawArray(int n, int value)
 {
+  if(n <= 0)
+  {
+    return NULL;
+  }
   if (mem_space_ == "DEFAULT")
   {
-    return new double[n];
+    double *tmp = new double[n];
+    std::fill(tmp, tmp + n, value);
```

Comment:
@CameronRutherford Can you remind me why we're `std::fill`ing for *every* allocation? Why can't the user `fill` if need be?

Reply:
￼We do not have to use `std::fill` for every allocation if we don't want. Instead, we could create two instances of this function. One that fills the array with a default value (using `std::fill`), and one that just initialises memory.

The reason I wanted to fill the memory with a value was to phase out usage of the internal testing function `createLocalBuffer`, however this may be too narrow of a use case to apply this change to all other parts of HiOp code that call this function.